src/borrow_tracker/tree_borrows/diagnostics.rs

```diff
@@ -28,7 +28,7 @@ impl fmt::Display for AccessCause {
             Self::Explicit(kind) => write!(f, "{kind}"),
             Self::Reborrow => write!(f, "reborrow"),
             Self::Dealloc => write!(f, "deallocation"),
-            Self::FnExit => write!(f, "protector release"),
+            Self::FnExit(kind) => write!(f, "protector release ({kind})"),
```

Comment:
```suggestion
            // These are dead code, since the protector release access itself can never
            // cause UB (while the protector is active, if some other access invalidates
            // further use of the protected tag, that is immediate UB).
            // Describing the cause of UB is the only time this function is called.
            Self::FnExit(_) => unreachable!("protector accesses can never be the source of UB"),
```
This way if someone out there happens to somehow trigger this, we'll know. :)

Reply:
Ah, turns out this code path is triggered, for instance it produces [this](https://github.com/rust-lang/miri/blob/2982438e8970b194fb4cdd2e022c3a4356c7dfec/tests/fail/tree_borrows/spurious_read.stderr#L27) diagnostic.

Reply:
I've added it, and all tests still passed.